file/src/main/java/org/apache/calcite/adapter/file/CsvEnumerator.java

```diff
@@ -314,22 +371,55 @@ static CSVReader openCsv(Source source) throws IOException {
         } catch (ParseException e) {
           return null;
         }
-      case STRING:
+      case VARCHAR:
       default:
         return string;
       }
     }
   }
 
+  private static RelDataType parseDecimalSqlType(JavaTypeFactory typeFactory, int precision,
+      int scale) {
+    if (scale < 0) {
```

Comment:
It seems like you should do some other checks here as well. For example, can precision be negative? What are the requirements for a relationship between precision and scale?

Reply:
Done. The following should be enough:
1. precision > 0: 0 has precision 1 and scale 0
2. scale >= 0
3. precision >= scale

Reply:
Do you want to update these to use Preconditions? I think it has convenience methods to use format strings (although I think it simplifies to %s for everything).

Reply:
Done.